x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/action/ShardFollowTaskReplicationTests.java

```diff
@@ -279,6 +279,33 @@ public void testRetryBulkShardOperations() throws Exception {
         }
     }
 
+    public void testOutOfOrderOnFollower() throws Exception {
```

Comment:
it's not about out of order but rather over recovery of ops you already have. I checked and I can't find any existing tests that cover recovery (which can cover this case). I think we should have some (i.e., rename this test and extend it to cover general peer recovery (with and without holes). We also need concurrent recovery + indexing.

Reply:
Yes, I'll do.